crates/fuel-core/src/query/message.rs

```diff
@@ -52,65 +56,75 @@ use std::borrow::Cow;
 #[cfg(test)]
 mod test;
 
-pub struct MessageQueryContext<'a>(pub &'a Database);
+pub trait MessageQueryData: Send + Sync {
+    fn message(&self, message_id: &MessageId) -> StorageResult<Message>;
+    fn owned_message_ids(
+        &self,
+        owner: &Address,
+        start_message_id: Option<MessageId>,
+        direction: IterDirection,
+    ) -> BoxedIter<StorageResult<MessageId>>;
```

Comment:
```suggestion
    ) -> BoxedIter<StorageResult<MessageId>>;
    
```

And for other methods too